src/Services/ZipCode.php

```diff
@@ -197,7 +197,7 @@ protected function fetchZipCodeAddress()
         $address = $this->parsedXML['consultaCEPResponse']['return'];
         $zipcode = preg_replace('/^([0-9]{5})([0-9]{3})$/', '${1}-${2}', $address['cep']);
         $complement = array_values(array_filter([
-            $address['complemento'], $address['complemento2']
+            isset($address['complemento']) ? $address['complemento'] : '', isset($address['complemento2']) ? $address['complemento2'] : ''
```

Comment:
@LucienCorreia sugestão para deixar o código um pouco mais simples, criar um novo método na classe que será responsável por tratar o complemento, exemplo:

```php
protected function getComplement($address)
{
     $complement = [];

    if (array_key_exists('complemento', $address)) {
        $complement[] = $address['complemento'];
    }

    if (array_key_exists('complemento2', $address)) {
        $complement[] = $address['complemento2'];
    }

    return $complement;
}
```

E mudar essa linha para usar esse novo método :)

```suggestion
$complement = $this->getComplement($address);
```